projects/04/fill/Fill.asm

```diff
@@ -6,9 +6,56 @@
 // Runs an infinite loop that listens to the keyboard input.
 // When a key is pressed (any key), the program blackens the screen,
 // i.e. writes "black" in every pixel;
-// the screen should remain fully black as long as the key is pressed. 
+// the screen should remain fully black as long as the key is pressed.
 // When no key is pressed, the program clears the screen, i.e. writes
 // "white" in every pixel;
 // the screen should remain fully clear as long as no key is pressed.
 
-// Put your code here.
\ No newline at end of file
+(START)
+@SCREEN
+D=A
+@R0
+M=D
+
+(CHANGECOLOUR)
+@KBD // check keyboard
+D=M
+
+@R1
+M=0 // default to white
+
+@MAKEDARK // change to black
+D;JNE // if not 0
+
+(CONT)
+@R1
+A=M
+D=A // get the colour
+
+@R0 // pointer to screen memory address
+A=M
+M=D // change to the colour
+
+@R0
+M=M+1 // store new pointer
+
+@24576

```

Comment:
It’s slightly naughty to use a literal `@24576` here rather than calculating it from the address of `@SCREEN` and the amount of memory used by the screen (8192 words), but I’m gonna let it go. It works!